.github/workflows/codeql-analysis.yml

```diff
@@ -40,32 +40,47 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@v2
+      with:
+        fetch-depth: '0'
+        token: ${{ secrets.BOT_TOKEN }}
+
+    - name: Set up JDK 17
+      uses: actions/setup-java@v1
+      with:
+        java-version: 17
+
+    - uses: actions/cache@v3
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Authenticate with Google Cloud
+      id: auth
+      uses: google-github-actions/auth@v2
+      with:
+        credentials_json: ${{ secrets.GCR_KEY }}
+    - name: Setup Google Cloud SDK
+      uses: google-github-actions/setup-gcloud@v2
+    - name: Configure GCR
+      run: |
+        gcloud auth configure-docker
```

Comment:
Not entirely sure why this has been added?

Reply:
The Autobuild step fails when it doesn't have these in place as it builds out the code. The cache isn't essential but makes the build (a bit) more performant so have left it there.